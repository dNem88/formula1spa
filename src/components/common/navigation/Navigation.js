import React, {useContext,  Fragment} from 'react';
import {Link} from 'react-router-dom'
import styles from './navigation.module.css'
import userContext from '../../../context/userContext'
import SignButton from '../authButton/SignButton'
function Navigation() {
    let context = useContext(userContext)
    console.log('context from navigation', context)
    return (
        <nav>
        {(context.isLogged === 'initial' && context.user === null) ? <p>Loading....</p> : 
            <Fragment>
                 <div className={styles['logo-container']}>
                    <img src={'https://www.formula1.com/etc/designs/fom-website/images/f1_logo.svg'}/>
                </div>
                <div className={styles['links-container']}>
                    <Link to='/latest'>Latest</Link>
                    <Link to='/video'>Video</Link>
                    <Link to='/schedule'>Schedule</Link>
                    <Link to='/standings'>Standings</Link>
                    <Link to='/drivers'>Drivers</Link>
                    <Link to='/teams'>Teams</Link>
                </div>
                <div className={styles['auth-container']}>
                    {context.isLogged === false ? <SignButton href={'/login'} textContent={'Sign in'}/> : null} 
                    {context.isLogged === false ? <SignButton href={'/register'} textContent={'Register'}/> : null} 
                    {context.isLogged === true ? <SignButton href={'/profile'} textContent={'Profile'}/> : null}
                </div>
            </Fragment>
        }
        </nav>
    )
}

export default Navigation
