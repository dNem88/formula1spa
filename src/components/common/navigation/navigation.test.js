import renderer from 'react-test-renderer';
import Navigation from './navigation';
import React from 'react'
import userContext from '../../../context/userContext'
import {BrowserRouter as Router} from 'react-router-dom'

describe('test Navigation component', () => {
  it('Should render logged in user Navigation', () => {
    let loggedInNavigation = renderer.create(
        <Router>
            <userContext.Provider value={{isLogged: true, user: {username: 'dani', id: '123'}}}>
                <Navigation/>
            </userContext.Provider>
        </Router>
    ).toJSON()
    expect(loggedInNavigation).toMatchSnapshot()
    expect(loggedInNavigation.type).toBe('nav')
    expect(loggedInNavigation.children[0].type).toBe('div')
    expect(loggedInNavigation.children[1].type).toBe('div')
    expect(loggedInNavigation.children[2].type).toBe('div')
     expect(loggedInNavigation.children[2].children[0].type).toBe('a')
     expect(loggedInNavigation.children[2].children[0].children[0]).toBe('Profile')
  })
  it('Should render guest user navigation', () => {
    let guestNavigation = renderer.create(
        <Router>
            <userContext.Provider value={{isLogged: false, user: null}}>
                <Navigation/>
            </userContext.Provider>
        </Router>
    ).toJSON()
    expect(guestNavigation).toMatchSnapshot()
    expect(guestNavigation.type).toBe('nav')
    expect(guestNavigation.children[0].type).toBe('div')
    expect(guestNavigation.children[1].type).toBe('div')
    expect(guestNavigation.children[2].type).toBe('div')
    expect(guestNavigation.children[2].children[0].type).toBe('a')
    expect(guestNavigation.children[2].children[0].children[0]).toBe('Sign in')
    expect(guestNavigation.children[2].children[1].children[0]).toBe('Register')
  })
})