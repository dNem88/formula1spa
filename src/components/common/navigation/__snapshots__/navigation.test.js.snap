// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Navigation component Should render guest user navigation 1`] = `
<nav
  className="nav"
>
  <div
    className="logo-container"
  >
    <img
      src="https://www.formula1.com/etc/designs/fom-website/images/f1_logo.svg"
    />
  </div>
  <div
    className="links-container"
  >
    <a
      href="/latest"
      onClick={[Function]}
    >
      Latest
    </a>
    <a
      href="/video"
      onClick={[Function]}
    >
      Video
    </a>
    <a
      href="/schedule"
      onClick={[Function]}
    >
      Schedule
    </a>
    <a
      href="/standings"
      onClick={[Function]}
    >
      Standings
    </a>
    <a
      href="/drivers"
      onClick={[Function]}
    >
      Drivers
    </a>
    <a
      href="/teams"
      onClick={[Function]}
    >
      Teams
    </a>
  </div>
  <div
    className="auth-container"
  >
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Sign in
    </a>
    <a
      className="link"
      href="/auth/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</nav>
`;

exports[`test Navigation component Should render logged in user Navigation 1`] = `
<nav
  className="nav"
>
  <div
    className="logo-container"
  >
    <img
      src="https://www.formula1.com/etc/designs/fom-website/images/f1_logo.svg"
    />
  </div>
  <div
    className="links-container"
  >
    <a
      href="/latest"
      onClick={[Function]}
    >
      Latest
    </a>
    <a
      href="/video"
      onClick={[Function]}
    >
      Video
    </a>
    <a
      href="/schedule"
      onClick={[Function]}
    >
      Schedule
    </a>
    <a
      href="/standings"
      onClick={[Function]}
    >
      Standings
    </a>
    <a
      href="/drivers"
      onClick={[Function]}
    >
      Drivers
    </a>
    <a
      href="/teams"
      onClick={[Function]}
    >
      Teams
    </a>
  </div>
  <div
    className="auth-container"
  >
    <a
      className="link"
      href="/auth/profile"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>
</nav>
`;
